src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -8262,6 +8262,41 @@ internal static FileStream CreateFileStream(string path, string streamName, File
             return new FileStream(handle, access);
         }
 
+        /// <summary>
+        /// Tries to create a file stream on a file.
+        /// </summary>
+        /// <param name="path">The fully-qualified path to the file.</param>
+        /// <param name="streamName">The name of the alternate data stream to open.</param>
+        /// <param name="mode">The FileMode of the file.</param>
+        /// <param name="access">The FileAccess of the file.</param>
+        /// <param name="share">The FileShare of the file.</param>
+        /// <param name="stream">A FileStream that can be used to interact with the file.</param>
+        /// <returns>true if the stream was successfully created, otherwise false.</returns>
+        internal static bool TryCreateFileStream(string path, string streamName, FileMode mode, FileAccess access, FileShare share, out FileStream stream)
```

Comment:
Both CreateFileStream and TryCreateFileStream use common code - I think we should use "dry" principle. 

Also please fix formatting issues for the code.